torch/distributed/tensor/placement_types.py

```diff
@@ -349,11 +324,6 @@ def _to_new_shard_dim(
 
         return new_tensor
 
-    def __eq__(self, other: object) -> bool:
-        if not isinstance(other, Shard):
-            return False
-        return self.dim == other.dim
-
     def __hash__(self) -> int:
```

Comment:
Same here, why not define repr and eq, and hash in C++ for consistency?

Reply:
hm, I agree that `__hash__` should be defined next to `__eq__`. I'll fix that. for `__repr__` I'm trying to force as little as possible to be done in C++, and it's fundamentally a Python thing.